x/ecocredit/server/core/query_batches.go

```diff
@@ -0,0 +1,45 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// Batches queries for all batches in the given credit class.
+func (k Keeper) Batches(ctx context.Context, request *v1beta1.QueryBatchesRequest) (*v1beta1.QueryBatchesResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+	project, err := k.stateStore.ProjectInfoStore().GetByName(ctx, request.ProjectId)
```

Comment:
a bit confusing that the ProjectId in the request refers to the unique project name (string), and not its DB id (uint64). But i guess it's fine thanks to the different types